mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/FileBasedPersistenceProvider.java

```diff
@@ -0,0 +1,85 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.master.persistence;
+
+import io.mantisrx.master.events.LifecycleEventPublisher;
+import io.mantisrx.master.events.LifecycleEventsProto;
+import io.mantisrx.master.jobcluster.IJobClusterMetadata;
+import io.mantisrx.master.jobcluster.job.IMantisJobMetadata;
+import io.mantisrx.server.master.store.FileBasedStore;
+import io.mantisrx.server.master.store.KeyValueStore;
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Simple File based storage provider. Intended mainly as a sample implementation for
+ * {@link IMantisPersistenceProvider} interface. This implementation is complete in its functionality, but, isn't
+ * expected to be scalable or performant for production loads.
+ * <P>This implementation uses <code>/tmp/MantisSpool/</code> as the spool directory. The directory is created
+ * if not present already. It will fail only if either a file with that name exists or if a directory with that
+ * name exists but isn't writable.</P>
+ */
+public class FileBasedPersistenceProvider extends KeyValueBasedPersistenceProvider {
+    private static final Logger logger = LoggerFactory.getLogger(FileBasedPersistenceProvider.class);
+    private static final LifecycleEventPublisher noopEventPublisher = new LifecycleEventPublisher() {
+        @Override
+        public void publishAuditEvent(LifecycleEventsProto.AuditEvent auditEvent) {
+        }
+
+        @Override
+        public void publishStatusEvent(LifecycleEventsProto.StatusEvent statusEvent) {
+        }
+
+        @Override
+        public void publishWorkerListChangedEvent(LifecycleEventsProto.WorkerListChangedEvent workerListChangedEvent) {
+        }
+    };
+
+    public FileBasedPersistenceProvider(boolean actualStorageProvider) {
+        this((actualStorageProvider) ? new FileBasedStore() : KeyValueStore.NO_OP,
+            noopEventPublisher);
+    }
+
+    public FileBasedPersistenceProvider(KeyValueStore sprovider, LifecycleEventPublisher publisher) {
+        super(sprovider, publisher);
+    }
+
+    public FileBasedPersistenceProvider(FileBasedStore sprovider) {
+        this(sprovider, noopEventPublisher);
+    }
+
+    public FileBasedPersistenceProvider(File stateDirectory, boolean eventPublisher) {
+        this(new FileBasedStore(stateDirectory), noopEventPublisher);
+    }
+
+    Optional<IJobClusterMetadata> loadJobCluster(String clusterName) throws IOException {
```

Comment:
is this an override?

Reply:
No, this is a class used only in tests and had a completely separate implementation earlier. 
There were these two helpful methods defined used only in tests. I chose to keep them and simplify the implementation a bit.

Reply:
nit: add a doc or testing annotation.